scalyr_agent/monitor_utils/k8s.py

```diff
@@ -127,9 +785,37 @@ def query_pod( self, namespace, pod ):
 
     def query_pods( self, namespace=None, filter=None ):
         """Wrapper to query all pods in a namespace, or across the entire cluster"""
-        query = '/api/v1/pods'
+        return self.query_objects( '/api/v1/pods', '/api/v1/namespaces/%s/pods', namespace, filter )
```

Comment:
If seems like namespace=None would be bad here.  Maybe we shouldn't have it default to None in the argument list?

Reply:
No, None is ok here and if None, then the entire cluster is queried rather than just the namespace.  This function is just a convenience wrapper that passes in the appropriate urls.  The None checking is performed in query_objects, rather than duplicating the exact same logic in each of these query_*s methods.  I've updated the comments of the methods to mention this.